src/main/java/duke/memo/data/TaskList.java

```diff
@@ -0,0 +1,54 @@
+package duke.memo.data;
+
+import duke.memo.exception.DukeException;
+import duke.memo.task.Deadline;
+import duke.memo.task.Event;
+import duke.memo.task.Task;
+import duke.memo.task.ToDo;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+
+public class TaskList extends ArrayList<Task> {
+
+    public TaskList() {
+        super();
+    }
+
+    /**
+     * Constructor for TaskList.
+     *
+     * @param taskList  list of tasks.
+     * @throws DukeException throw if there is a problem in the command.
+     */
+    public TaskList(ArrayList<String> taskList) throws DukeException {
+        super();
+        Iterator<String> iter = taskList.iterator();
+        while (iter.hasNext()) {
+            add(readLog(iter.next()));
+        }
+    }
+
+    /**
+     * Convert raw log string to a task object.
+     *
+     * @param log  Raw log.
+     * @return Task return the parsed log
+     * @throws DukeException Throw if there is a problem in the log.
+     */
+    public Task readLog(String log) throws DukeException {
```

Comment:
Not much issues with the code, but perhaps the readLog method would be more appropriate to be placed in the Parser class, since you are parsing strings